src/libprojectM/Renderer/Renderer.cpp

```diff
@@ -229,6 +236,16 @@ void Renderer::ResetTextures()
 void Renderer::SetupPass1(const Pipeline& pipeline, const PipelineContext& pipelineContext)
 {
 	totalframes++;
+	this->currentTime = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+	milliseconds ms = std::chrono::duration_cast<std::chrono::milliseconds>(this->currentTime - this->lastTime);
+	double diff = ms.count();
+	if (diff >= 250)
+	{
```

Comment:
comment explaining this block would be nice

Reply:
There is already a total frame count (totalframes). The idea is to track from the initial run until we reach 250 milliseconds (1/4 of a second). To estimate the FPS I multiply the totalframes rendered in 1/4 of a second by 4. That way we get an update every 1/4 of a second.

Reply:
I will put in a comment block to explain the logic.